server/app.js

```diff
@@ -1,15 +1,148 @@
 const express = require("express");
-
+const mongoose = require("mongoose");
 const app = express();
+const cors = require("cors");
+const Students = require("./models/student");
+const Course = require("./models/course");
+const { restart } = require("nodemon");
 
 app.use(express.json());
+app.use(cors());
+
+//Get Endpoints
+
+//get a single course
+app.get("/course/:courseId", (req, res) => {
+  const { id } = req.params;
+  Course.findById(id).then((courses) => {
+    res.status(200);
+    res.json(course);
+  });
+});
+
+//get all courses
+app.get("/course", (req, res) => {
+  Course.find().then((course) => {
+    res.status(200);
+    res.json(course);
+  });
+});
+
+//get all Students in a course
+app.get("/courses/:courseId/students", (req, res) => {
+  Students.find().then((students) => {
+    res.status(200);
+    res.json(students);
+  });
+});
+
+//get a single Student from a specific course
+app.get("/courses/:courseId/studentsId", (req, res) => {
+  const { studentId } = req.params;
+  Students.findById(studentId).then((students) => {
+    res.status(200);
+    res.json(students);
+  });
+});
```

Comment:
since there is no colon in front of "studentsId" in the query `studentId` will always be undefined. Also, you're not using the `courseId`  query parameter. 

I talked about that task in another review, sorry.
https://github.com/PhilippAubert/Api-From-Scratch/pull/1#discussion_r615675213